### PR TITLE
IDENTITY-6456: Fix tenantID is not correctly assigned for super tenan…

### PR DESCRIPTION
**Pull request by hasinidilanka**

### CODE REVIEW

**Comment by hasinidilanka** [comment]
@omindu @darshanasbg @isharak @mefarazath

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public String jwks() {
             jwksJson.put("keys", jwksKeyArray);
         } catch (Exception e) {
             String errorMesage = "Error while generating the keyset";
-            log.error(errorMesage);
+            log.error(errorMesage , e);
```
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java]: Unnecessary space after `errorMesage`

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ public String jwks() {
             jwksJson.put("keys", jwksKeyArray);
         } catch (Exception e) {
             String errorMesage = "Error while generating the keyset";
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java]: Can we add some more context information such tenantDomain?

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java [hunk 3]
```diff
@@ -114,8 +114,8 @@ public String jwks() {
             jwksKeyArray.put(jwksKeys);
             jwksJson.put("keys", jwksKeyArray);
         } catch (Exception e) {
-            String errorMesage = "Error while generating the keyset";
-            log.error(errorMesage);
+            String errorMesage = "Error while generating the keyset for " + tenantDomain + "tenant domain.";
```
Comment (DMHP) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java]: Please put a space between +tenantDomain+ and " tenant domain."

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java [hunk 4]
```diff
@@ -114,8 +114,8 @@ public String jwks() {
             jwksKeyArray.put(jwksKeys);
             jwksJson.put("keys", jwksKeyArray);
         } catch (Exception e) {
-            String errorMesage = "Error while generating the keyset";
-            log.error(errorMesage);
+            String errorMesage = "Error while generating the keyset for " + tenantDomain + " tenant domain.";
```
Comment (Abilashini) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java]: Can be improved as "Error while generating the keyset for tenant domain: " + tenantDomain

Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java]: Fixed in master

**Pull request merged**